src/odemis/util/peak.py

```diff
@@ -318,14 +368,110 @@ def estimateFitTime(self, data):
+
+    """
+    amplitude_e = (amplitude * pos ** 2) / constant
+    width_e = constant * ((1 / (pos - width / 2)) - (1 / (pos + width / 2)))
+    pos_e = constant / pos
+    return pos_e, width_e, amplitude_e
+
+
+def peak_to_wavelength(pos_e, width_e, amplitude_e):
+    """
+    Converts the peaks to space domain.
+    Args:
+        pos_e: the center position of the peak in 'eV'
+        width_e: the width of the peak in 'eV'
+        amplitude_e: the amplitude of the peak in energy domain
+
+    Returns: peak parameters as (pos, width, amplitude) in space domain
+
+    """
+    width = constant * ((1 / (pos_e - width_e / 2)) - (1 / (pos_e + width_e / 2)))
+    pos = constant / pos_e
+    amplitude = (amplitude_e * constant) / pos ** 2  # pos should be in space domain for the reverse Jacobian
+    return pos, width, amplitude
+
+
+def calc_width(initial_width, wavelength):
+    """
+    Calculates a rough estimation of the width by multiplying the initial width by the 1/10 of the wavelength range.
+    Args:
+        initial_width: a parameter which will be multiplied by the ratio peak width of wavelength
+        wavelength: the list of wavelengths in "m" or "px
+
+    Returns: peak width estimation
+
+    """
+    peak_width_ratio = (wavelength[-1] - wavelength[0]) / 10
+    return initial_width * peak_width_ratio
+
+
+def applyJacobian_x(wavelength):
+    """
+    Converts the wavelength from 'm' (space domain) to 'eV' (energy domain).
+    Args:
+        wavelength(1d array of floats): the list of wavelengths in 'm'
+
+    Returns:
+         energy (1d array of floats): wavelength in 'eV'
+
+    """
+    wavelength = numpy.asarray(wavelength)
+    energy = constant / wavelength
+    return energy
+
+
+def applyJacobian_y(wavelength, spectrum):
+    """
+    Applies Jacobian transformation to the spectrum.
+    Args:
+        wavelength(1d array of floats): wavelength values corresponding to the given spectrum in 'm'
+        spectrum(1d array of floats): data representing the spectrum in space domain
+
+    Returns:
+         spectra_energy (1d array of floats): data representing the spectrum in energy domain
+
+    """
+    wavelength = numpy.asarray(wavelength)
+    spectra_energy = spectrum * (wavelength ** 2) / constant
+    return spectra_energy
+
+
+def apply_reverseJacobian_y(wavelength, curve_energy):
+    """
+    Applied inverse Jacobian transformation to the curve.
+    Args:
+        wavelength (1d array of floats): wavelength values in 'eV'
+        curve_energy (1d array of floats): dataset of points representing the curve in energy domain
+
+    Returns:
+        curve_space (1d array of floats): dataset of points representing the curve in space domain
+
+    """
+    wavelength = numpy.asarray(wavelength)
+    curve_space = (curve_energy * constant) / (wavelength ** 2)
+    return curve_space
+
+
+def Curve(wavelength, peak_parameters, offset, type='gaussian_space'):
```

Comment:
start with a capital, while being a non protected member? Maybe I am being a bit hard on method naming (I ain't a saint in that either)